sympy/matrices/matrices.py

```diff
@@ -3981,3 +3981,69 @@ def classof(A, B):
 def a2idx(j, n=None):
     from sympy.matrices.common import a2idx as a2idx_
     return a2idx_(j, n)
+
+def upper(a, diag=0):
+    from sympy import Matrix
```

Comment:
I think that you should inspect of the class of `a` instead of forcing everything as Matrix, because your code forces every matrix into `MutableDenseMatrix`

Reply:
Also there shouldn't be code before a docstring.

Reply:
ok I have added imports after docstring.